### PR TITLE
core/translate: fix USING/NATURAL qualified-column ambiguity with alias collisions

### PR DESCRIPTION
**Pull request by Pavan-Nambi**
# NOTICE:

### CODE REVIEW

File: testing/sqltests/tests/update-from.sqltest [hunk 1]
```diff
@@ -1864,6 +1864,146 @@ expect error {
...
+     WHERE EXISTS (SELECT 1 WHERE x.b = 7);
+}
+expect error {
+    ambiguous column name: x.b
+}
+
+# Positive sanity check: the qualified-ambiguity fix above must not
+# over-reject a legitimate USING join when the source aliases are unique and
+# the qualified reference points unambiguously at a single FROM table.
+test update-from-target-alias-collision-fix-still-allows-unambiguous-using {
```
Comment (Pavan-Nambi) [inline testing/sqltests/tests/update-from.sqltest:1970]: note that this  is not a regression test, it was a generated when i asked claude to generate some more edge cases it can think of

File: testing/sqltests/tests/update-from.sqltest [hunk 2]
```diff
@@ -1864,6 +1864,146 @@ expect error {
...
+     WHERE l.a = d.a;
+    SELECT * FROM dst;
+}
+expect {
+    5|5
+}
+
+# Multi-USING chain with right-side aliases colliding with the UPDATE target.
+# Ensure duplicate-identifier ambiguity still triggers before column resolution.
+test update-from-target-alias-collision-using-chain-self-aliased {
```
Comment (Pavan-Nambi) [inline testing/sqltests/tests/update-from.sqltest:1989]: this is not a regression test either.

**Comment by penberg** [comment]
This pull request has been marked as stale due to inactivity. It will be closed in 7 days if no further activity occurs.

**Comment by penberg** [comment]
This pull request has been closed due to inactivity. Please feel free to reopen it if you have further updates.
